tests/test_style.py

```diff
@@ -485,6 +491,8 @@ def test_fetch_private_github_urls(tmp_path):
         missing = "thing"{SUGGESTION_END}
         """
     )
+    # 'Basic eHh4Og==' is b64 encoding of "Basic xxx:"
+    assert responses.calls[0].request.headers["Authorization"] == "Basic eHh4Og=="
     project.flake8(offline=True).assert_no_errors()
 
 
```

Comment:
> API _Has some, and some doctests, but might need more - suggestions welcomed_

I would add one more test function `test_parsing_github_urls_with_token()` to cover these cases:

- URL with a literal token
- URL with a token in an environment variable
- URL with a token in the query string

Then you could test the output of the `credentials` property on the `GitHubURL` class.

See `test_parsing_github_urls()` and `test_parsing_python_package_urls()` (below in this file) as references.